File: bench/hl/sample.go
package main

import "fmt"

// 反引号原始字符串的词法陷阱(T67 语料):内部反斜杠不转义
const pattern = `C:\Users\测试\路径`

// 打印中文问候,含中文注释
func greet(name string) string {
	return fmt.Sprintf("你好,%s!欢迎使用 mdvn", name)
}

func main() {
	fmt.Println(greet("世界"))
	fmt.Println(pattern)

	// 多行反引号原始字符串,常用于 SQL/正则等场景
	sql := `SELECT *
FROM 用户表
WHERE 名字 = '张三'`
	fmt.Println(sql)
}

File: bench/hl/sample.rs
// Rust 样例:r#"..."# 原始字符串与生命周期标注的词法陷阱(T67 语料)
// 'a 生命周期极易被误判成字符字面量的开头,这是 Rust 高亮最经典的坑。

// 带生命周期标注的结构体,持有一个字符串切片引用
struct Greeter<'a> {
    name: &'a str,
}

impl<'a> Greeter<'a> {
    // 构造函数,返回带生命周期的实例
    fn new(name: &'a str) -> Greeter<'a> {
        Greeter { name }
    }

    // 生成中文问候语
    fn greet(&self) -> String {
        format!("你好,{}!欢迎使用 mdvn", self.name)
    }
}

fn main() {
    // r#"..."# 原始字符串,内部可以直接写双引号不用转义
    let pattern = r#"路径示例: "C:\Users\测试""#;
    let g = Greeter::new("世界");
    println!("{}", g.greet());
    println!("{}", pattern);

    // 真正的字符字面量,和生命周期标注 'a 的写法只差一个字符
    let ch: char = 'x';
    println!("{}", ch);
}
